tokio/src/sync/semaphore.rs

```diff
@@ -102,6 +76,54 @@ use std::sync::Arc;
 /// }
 /// ```
 ///
+/// Limit the number of incoming requests being handled at the same time.
+///
+/// Similar to limiting the number of simultaneously opened files, network handles
+/// are a limited resource. Allowing an unbounded amount of requests to be processed
+/// could result in a denial-of-service, among many other issues.
+///
+/// However, in contrast to the file example, this example uses an `Arc<Semaphore>`.
+/// Before we spawn a new task to process a request, we must acquire a permit from
+/// this semaphore, in order to limit usage of system resources (like file descriptors,
+/// memory usage, CPU time, etc.). Once acquired, a new task is spawned; and once
+/// finished, the permit is dropped inside of the task to allow others to spawn.
+/// Permits must be acquired via [`Semaphore::acquire_owned`], in order to be
+/// movable across the task boundary. Contrastingly, if the semaphore were a global,
+/// as in the file-limiting example, only ['Semaphore::acquire'] would be necessary.
+/// This is because a `Permit<'static>` would be returned, which is movable across
+/// task boundaries, since the semaphore lives forever.
+///
+/// ```no_run
+/// use std::sync::Arc;
+/// use tokio::sync::Semaphore;
+/// use tokio::net::{TcpListener, TcpStream};
+///
+/// #[tokio::main]
+/// async fn main() -> Result<(), Box<dyn std::error::Error>> {
+///     let semaphore = Arc::new(Semaphore::new(3));
+///     let listener = TcpListener::bind("127.0.0.1:8080").await?;
+///
+///     loop {
+///         // Acquire permit before accepting, to avoid needlessly taking up a (finite) file descriptor
```

Comment:
I don't think using finite like this is grammatically correct.
```suggestion
///         // Acquire permit before accepting the next socket.
```

Reply:
As far as I can tell, using Oxford Languages as a casual source, it is grammatically correct -- though it would be less flourishy if the parens were removed. 

See: 
![finite-ex](https://github.com/tokio-rs/tokio/assets/62392572/2fa1ab86-2bb1-4337-b333-2681df7f634e)

Are we also removing the explanation/justification for acquiring a permit before accepting a socket? The document is always changing, so I understand if this fits better within the whole.

Reply:
Sorry, let me correct myself. I agree that it is grammatically correct, but I don't think using finite in this way has the meaning you intended.

I think you wanted to say that file descriptors are a finite resource, correct? However, you're saying that the file descriptor itself is finite, which is a different statement.

Reply:
You're right. It was a sloppy error on my part.